### ISSUE TITLE
Terratest assert.Equal is showing NOT EQUAL even though expected & actual values are same.

### ISSUE THREAD

**Issue by desaikeyurb** [gruntwork-io/terratest#870]
I am running Terratest against terraform code available on azure. It is successfully executing terraform code and able to create the infra, validate my test case and destroying everything after validation. However, while validating the test case, assert.Equal is not working as expected. 
I have written simple test case(to start with) to compare expected & actual VM size. Since it is not working added two more test cases as below. So total 3 test cases are there as below:

1.	Comparing VM size: assert.Equal is showing NOT EQUAL even though expected & actual values are same. ![Assert Equal failing_Code](https://user-images.githubusercontent.com/78408265/115395707-cd6be780-a201-11eb-94db-a239af5ee1fb.png). (Printed/logged the type - both are of same type "compute.VirtualMachineSizeTypes").
2.	Since first test case is failing, I have formatted and return both the VM size in string and then did assert.Equal, however it is not at all scanning - NOT GIVING ANY RESULT
3.	Hard coded two string and tried to compare using assert.Equal however this is not scanning either. - NOT GIVING ANY RESULT.

Please if anyone can suggest on above three scenarios, ideally two only -
assert.Equal is not giving expected result.
assert.Equal is not at all scanning and not giving any result.

Go version: go1.16.3 linux/amd64
Terraform version: Terraform v0.12.26

**Comment by desaikeyurb** [comment]
1.	Comparing VM size: assert.Equal is showing NOT EQUAL even though expected & actual values are same. (Printed/logged the type - both are of same type "compute.VirtualMachineSizeTypes").
2.	Since first test case is failing, I have formatted and return both the VM size in string  and then did assert.Equal, however it is not at all scanning.
3.	Hard coded two string and tried to compare using assert.Equal however this is not scanning either.

**Comment by desaikeyurb** [comment]
Can anyone please help me on above.

**Comment by yorinasub17** [comment]
Hi, I don't quite understand what you mean by "This is not scanning," but note that `assert.Equal` will not return any log messages if it passes the assertion, so that might be what you are seeing?

Also, it would help if you can share the actual result of the `assert` failure in the logs. My best guess is that you have mismatch between the version of the Azure SDK you are using in your direct dependency, and the one `terratest` uses. This could happen if your import paths are not matched. Double check [the API version](https://github.com/gruntwork-io/terratest/blob/master/modules/azure/compute.go#L6) and make sure they match.

**Comment by desaikeyurb** [comment]
Thanks @yorinasub17 for your response. Azure SDK versions were same "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute". However the issue got resolved now. Some issue was with setup & access. I have reinstalled require package, modules and dependencies and got some firewall related issues fixed with IT team. 
Also, thanks for confirming that assert.Equal will not return any log messages if it passes the assertion, it just move ahead.
